lib/miner-info.js

```diff
@@ -23,11 +23,17 @@ const smartContractClient = new ethers.Contract(
  * @returns {Promise<string>} The peer ID of the miner.
  */
 export async function getIndexProviderPeerId(minerId, { maxAttempts = 5 } = {}) {
-  const { peerId, source } = await getPeerId(minerId, smartContractClient, {
-    rpcUrl: RPC_URL,
-    rpcAuth: RPC_AUTH,
-    maxAttempts,
-  })
-  console.log(`Peer ID fetched from ${source}.`)
-  return peerId
+  try {
+    const { peerId, source } = await getPeerId(minerId, smartContractClient, {
+      rpcUrl: RPC_URL,
+      rpcAuth: RPC_AUTH,
+      maxAttempts,
+    })
+    console.log(`Peer ID fetched from ${source}.`)
+    return peerId
+  } catch (err) {
+    const error = new Error(`Failed to get peer Id for miner ${minerId}.`, { cause: err })
+    console.error(error)
+    throw error
+  }
```

Comment:
I believe this will repeat the same sentence twice, since we are already wrapping the low-level error cause into another error mentioning miner ID.

https://github.com/CheckerNetwork/index-provider-peer-id/blob/42f0e1c6b5ab681aa7a998a104b43d5c2f269249/index.js#L44-L48

Suggested fix:

```suggestion
  } catch (err) {
    console.error(err)
    throw err
  }
```

Is this catch block actually necessary? Don't we have a top-level error handler that will print the error details?

@NikolasHaimerl can you please tweak `manual-check.js` to fail in this step and show us the console output.

Reply:
I added the try catch block because that was the behaviour we had in the preceding function that fetches the peer id: https://github.com/CheckerNetwork/spark-checker/blob/main/lib/miner-info.js#L84

Should we deviate from that behaviour?

Reply:
Addressed in c4c6f30